test/jdk/java/util/concurrent/forkjoin/AsyncShutdownNow.java

```diff
@@ -138,41 +109,79 @@ public void testTimedFutureGet(ExecutorService executor) throws Exception {
     @Test(dataProvider = "executors")
     public void testInvokeAll(ExecutorService executor) throws Exception {
         System.out.format("testInvokeAll: %s%n", executor);
-        scheduleShutdownNow(executor, 5);
-        try {
-            // execute long running tasks
+        try (executor) {
+            // shutdownNow when main thread waits in ForkJoinTask.quietlyJoin
+            onWait("java.util.concurrent.ForkJoinTask.quietlyJoin", executor::shutdownNow);
             List<Future<Void>> futures = executor.invokeAll(List.of(SLEEP_FOR_A_DAY, SLEEP_FOR_A_DAY));
             for (Future<Void> f : futures) {
                 assertTrue(f.isDone());
                 try {
                     Object result = f.get();
-                    assertTrue(false);
+                    fail();
                 } catch (ExecutionException | CancellationException e) {
                     // expected
                 }
             }
-        } finally {
-            executor.shutdown();
         }
     }
 
     /**
      * Test shutdownNow with thread blocked in invokeAny.
      */
-    @Test(dataProvider = "executors")
+    @Test(dataProvider = "executors", enabled = false)
```

Comment:
Should we file a JBS to re-enable this so that we don't forget?

Reply:
> Should we file a JBS to re-enable this so that we don't forget?

I'll add a note to JDK-8286352 or JDK-8288899, depending on which one is used.